cli/src/template_builder.rs

```diff
@@ -1063,6 +1064,65 @@ fn builtin_string_methods<'a, L: TemplateLanguage<'a> + ?Sized>()
             Ok(out_property.into_dyn_wrapped())
         },
     );
+    map.insert(
+        "replace",
+        |language, diagnostics, build_ctx, self_property, function| {
+            let ([pattern_node, replacement_node], [limit_node]) = function.expect_arguments()?;
+            let pattern = template_parser::expect_string_pattern(pattern_node)?;
+            let replacement_property =
+                expect_stringify_expression(language, diagnostics, build_ctx, replacement_node)?;
+
+            let is_regex = matches!(pattern, StringPattern::Regex(_) | StringPattern::RegexI(_));
+            let regex = pattern.to_regex();
+
+            let do_replace = move |haystack: &str, replace: &str, limit: Option<usize>| {
```

Comment:
nit: we no longer need `limit` to be an `Option`. There are two separate callers who pass `Some` or `None` unconditionally.